syntax = "proto3";

package FoodItemProto;

// Define the food categories enum
enum FoodCategory {
  UNKNOWN    = 0; // Proto3 requires first enum value to be zero
  UNPACKAGED = 1;
  PACKAGED   = 2;
  // Add categories here
}

// Message for representing dates (year, month, day)
message Date {
  int32 year  = 1;
  int32 month = 2;
  int32 day   = 3;
}

// Main FoodItem message
message FoodItem {
  int32 id              = 1;
  string image_path     = 2; // Storing path as string
  string name           = 3;
  FoodCategory category = 4;
  Date scan_date        = 5;
  Date expiration_date  = 6;
  int32 quantity        = 7;
}
